compiler/rustc_lint/src/builtin.rs

```diff
@@ -2804,10 +2803,48 @@ declare_lint! {
     "named labels in inline assembly",
 }
 
-declare_lint_pass!(NamedAsmLabels => [NAMED_ASM_LABELS]);
+declare_lint! {
+    /// The `binary_asm_labels` lint detects the use of numeric labels containing only binary
+    /// digits in the inline `asm!` macro.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,compile_fail
+    /// # #![feature(asm_experimental_arch)]
+    /// use std::arch::asm;
+    ///
+    /// fn main() {
+    ///     unsafe {
+    ///         asm!("0: bar");
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// A bug in LLVM causes this code to fail
```

Comment:
I would like us to expand on this explanation a bit, and ideally link to the original bug tracker here, unless we start to do so from rust by example.

Reply:
I forgot to expand this further, whoops.  I think I went to go get the bug link and got completely distracted along the way :sweat_smile: 

The Reference link below does also expand on it more and have the bug tracker link, but I think it's useful to duplicate a little bit of that here.